src/Headers.php

```diff
@@ -152,7 +152,7 @@ public function setEncoding($encoding)
     {
```

Comment:
I started to think all this function is a bad solution for proper header encoding. Header encoding should be done by each particular case.

Could we deprecate/remove this function? /cc @weierophinney